### PR TITLE
Update Managed Identity Samples with Logging

### PR DESCRIPTION
**Pull request by wsugarman**
Slightly refactor the managed identity samples to include diagnostic logging. Also used some newer syntax to (hopefully) simplify the sample code.

### CODE REVIEW

File: samples/ManagedIdentitySample/DTFx.AzureStorage v2.x/ManagedIdentity.AzStorageV2.csproj [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    <TargetFramework>net8.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Azure.Identity" Version="1.12.0" />
+    <PackageReference Include="Microsoft.Azure.DurableTask.AzureStorage" Version="2.0.0-rc.3" />
+    <PackageReference Include="Microsoft.Extensions.Azure" Version="1.7.4" />
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="6.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="6.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="6.0.0" />
```
Comment (cgillum) [inline samples/ManagedIdentitySample/DTFx.AzureStorage v2.x/ManagedIdentity.AzStorageV2.csproj:16]: Given that the same targets `net8.0`, should we update these nuget package versions to be the 8.0.x equivalents?

**Review by cgillum — COMMENTED** [review]
I love these improvements. Just one question below.

File: samples/ManagedIdentitySample/DTFx.AzureStorage v2.x/ManagedIdentity.AzStorageV2.csproj (hunk 1 above)
Comment (wsugarman) [reply samples/ManagedIdentitySample/DTFx.AzureStorage v2.x/ManagedIdentity.AzStorageV2.csproj:16]: Oops! That's what I get for copying 😅

Comment (wsugarman) [reply samples/ManagedIdentitySample/DTFx.AzureStorage v2.x/ManagedIdentity.AzStorageV2.csproj]: Updated with some additional refactoring

**Review by cgillum — APPROVED** [review]
LGTM!

**Pull request merged**